lib/stages/bufferSend.cpp

```diff
@@ -131,12 +136,15 @@ void bufferSend::main_thread() {
                 continue;
             }
             DEBUG2("Sent frame: {:d}", n_sent);
-            INFO("Sent frame: {:s}[{:d}] to {:s}:{:d}", buf->buffer_name, frame_id, server_ip,
-                 server_port);
+            DEBUG("Sent frame: {:s}[{:d}] to {:s}:{:d}", buf->buffer_name, frame_id, server_ip,
+                  server_port);
 
         } else {
-            WARN("Dropping frame {:s}[{:d}], because connection to {:s}:{:d} is down.",
-                 buf->buffer_name, frame_id, server_ip, server_port);
+            // Wait for connection and block
+            INFO("Waiting for connection to {:s}:{:d}...", server_ip, server_port);
+            std::unique_lock<std::mutex> connection_lock(connection_state_mutex);
+            connection_state_cv.wait(connection_lock, [&]() {return (true && connected);});
```

Comment:
Why does this return `true && connected` isn't that the same as `connected`?

Reply:
Also there is no way for this wait to break when `stop_thread == true`.   I think you should make this a `wait_for` with a timeout of say 1 second, and check the status of `stop_thread` as well.   Then make sure if the stop_thread condition is true break this loop so you can trigger the notify for the connection thread when calling `close_connection()` at the end of this function.

Reply:
ok, thanks for the feedback. I made those changes. The reason it was `true && connected` was that `connected` is an atomic type so I had to evaluate it somehow in that context.